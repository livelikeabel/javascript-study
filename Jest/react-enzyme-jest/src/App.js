import React from 'react';
import logo from './logo.svg';
import './App.css';

const Test = ({ text }) => <div>{text}</div>

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <h1>Welcome to React</h1>
      </header>
      <Test text="Some title" />
      <ul className="hi">
        <li>hi</li>
        <li>d</li>
        <li>2d</li>
      </ul>
    </div>
  );
}

export class Link extends React.Component {
  render() {
    return this.props.hide ? null : (<a href={this.props.address}>Click</a>)
  }
}

export default App;
