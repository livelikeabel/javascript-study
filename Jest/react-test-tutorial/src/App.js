import React, {Component} from 'react';
import Counter from './components/Counter'
import NameForm from './components/NameForm'
import NameList from './components/NameList'

class App extends Component {
  state = {
    names: ['abel', '의성']
  }
  
  onInsert = (name) => {
    this.setState(({names})=>({names: names.concat(name)}));
  }
  
  render() {
    return (
      <div>
          <Counter/>
          <hr />
          <h1>이름 목록</h1>
          <NameForm onInsert={this.onInsert}/>
          <NameList names={this.state.names} />
      </div>
    );
  }
}

export default App;
