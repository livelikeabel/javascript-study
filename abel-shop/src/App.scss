.App {
  width: 950px;
  margin: 0 auto;
  padding: 70px 0 0;
}