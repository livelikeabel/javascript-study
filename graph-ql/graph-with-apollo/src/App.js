import React from 'react';
import ApolloClient from 'apollo-boost';
import { ApolloProvider } from 'react-apollo';
import Recipes from './Recipes';

const client = new ApolloClient({
  uri: "http://localhost:4000/"
});

function App() {
  return (
    <ApolloProvider client={client}>
      Hello
     <Recipes/>
    </ApolloProvider>
  )
}

export default App;
