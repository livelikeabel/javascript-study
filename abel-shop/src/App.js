import React, { useState, createContext } from 'react';
import { BrowserRouter as Router, Route } from "react-router-dom";
import Main from './pages/main';
import Products from './pages/products';
import Wishlist from './pages/wishlist';

import './App.scss';

export const ShopContext = createContext();

const App = () => {
  const [wishlist, setWishlist] = useState([]);

  const globalState = { wishlist, setWishlist }

  return (
    <>
      <ShopContext.Provider value={globalState}>
        <Router>
          <div className="App">
            <Route exact path="/" component={Main} />
            <Route path="/products" component={Products} />
            <Route path="/wishlist" component={Wishlist} />
          </div>
        </Router>
      </ShopContext.Provider>
    </>
  );
};

export default App;
