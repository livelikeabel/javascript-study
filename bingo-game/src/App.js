import React, { Component } from 'react';
import { connect } from 'react-redux';
import BingoTable from './components/BingoTable';
import { test } from './reducers/bingo';
import './App.scss';

class App extends Component {
  render() {
    return (
      <div className="App">
        <BingoTable player={'player1'}/>
        <BingoTable player={'player2'}/>
      </div>
    );
  }
}

const mapStateToProps = ({ bingo: { player1 } }) => ({ player1 });
const mapDispatchToProps = { test }

export default connect(mapStateToProps, mapDispatchToProps)(App);
