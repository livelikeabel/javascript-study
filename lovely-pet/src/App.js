import React,{ Component } from 'react';
import { req } from './api';
import Card from './components/Card';
import './App.css';

class App extends Component {

  state = { cat: [], dog: [] };

  componentDidMount() {
    req('http://localhost:8080/dog').then(res => {
      console.log(res)
      this.setState({ dog: res })
    });
  }

  _renderCat() {
    if(this.state.dog.length === 0 ) return;
    return this.state.dog.map((data) => {
      return <Card {...data} key={data._id} />
    })
  }

  render() {
    return (
      <div className="App">
        냥이와 댕댕
        {/* {this.state.cat.length && this._renderCat()} */}
        {this.state.dog.length && this._renderCat()}
      </div>
    );
  }
}

export default App;
