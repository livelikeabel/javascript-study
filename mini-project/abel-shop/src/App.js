import React, { useState, createContext } from 'react';
import { BrowserRouter as Router, Route } from "react-router-dom";
import Main from './pages/main';
import Products from './pages/products';
import Wishlist from './pages/wishlist';

import './App.scss';

export const ShopContext = createContext();

const App = () => {
  const [wishlist, setWishlist] = useState([]);

  const globalState = { wishlist, setWishlist }

  // 리팩토링 한다면, 읽는 사람을 위해 wishlist의 initialstate 또는 데이터의 구조를 상수로 만들 것이다.
  return (
    <>
      <ShopContext.Provider value={globalState}>
        <Router>
          <div className="App">
            <Route exact path="/" component={Main} />
            <Route path="/products" component={Products} />
            <Route path="/wishlist" component={Wishlist} />
          </div>
        </Router>
      </ShopContext.Provider>
    </>
  );
};

export default App;
