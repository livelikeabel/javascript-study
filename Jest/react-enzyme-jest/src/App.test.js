import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() })

describe('<App />', () => {
  const wrapper = shallow(<App />)
  it("should contain 1 p element", () => {
    expect(wrapper.find('a').text()).toBe('Learn React')
  })
})