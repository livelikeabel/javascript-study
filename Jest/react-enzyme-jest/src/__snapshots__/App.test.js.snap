// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <h1>
      Welcome to React
    </h1>
  </header>
  <Test
    text="Some title"
  />
  <ul
    className="hi"
  >
    <li>
      hi
    </li>
    <li>
      d
    </li>
    <li>
      2d
    </li>
  </ul>
</div>
`;
